ci_scripts/check_category_linking.rb

```diff
@@ -22,9 +22,17 @@
 all_headers = (Dir.glob("Stripe/PublicHeaders/*.h") + Dir.glob("Stripe/*.h")).map do |h|
   filename = File.basename(h)
 end
-categories = (all_headers.select do |h|
-  h.include?("+") && !h.include?("Private") && !h.include?("Fabric")
-end).sort
+
+def is_category(h)
+  if h.include?("+")
+    return !h.include?("Private") && !h.include?("Fabric")
```

Comment:
I'm trying to understand what this category linking is doing and looking at https://medium.com/ios-os-x-development/categories-in-static-libraries-78e41f8ddb96

So, the problem is that whenever we have an implementation file that doesn't implement any classes, we have to manually link it? Which is why all of our `+Private.h` headers are fine? If that's it, then maybe this script needs to be updated to look at implementations instead of headers. Then we can get rid of these exceptions.

Backing up from that, it seems like asking developers to add `-ObjC` to their build flags is pretty standard and would let us avoid this problem. Our check script feels brittle and this problem is difficult to notice during our day-to-day development. I guess the hard part would be transitioning to this model. Maybe we can craft a check on initialization that would throw an assertion error.